service-base/src/main/java/org/eclipse/hono/service/registration/RegistrationHttpEndpoint.java

```diff
@@ -109,57 +117,41 @@ private void doGetDevice(final RoutingContext ctx) {
-            HttpUtils.badRequest(ctx, "body does not contain a valid JSON object");
-        }
+
+            registerDevice(ctx, ctx.get(KEY_REQUEST_BODY));
     }
 
     private void registerDevice(final RoutingContext ctx, final JsonObject payload) {
 
         if (payload == null) {
             HttpUtils.badRequest(ctx, "missing body");
         } else {
-            final Object deviceId = payload.remove(RegistrationConstants.FIELD_PAYLOAD_DEVICE_ID);
-            if (deviceId == null) {
-                HttpUtils.badRequest(ctx, String.format("'%s' param is required",
-                        RegistrationConstants.FIELD_PAYLOAD_DEVICE_ID));
-            } else if (!(deviceId instanceof String)) {
-                HttpUtils.badRequest(ctx, String.format("'%s' must be a string",
-                        RegistrationConstants.FIELD_PAYLOAD_DEVICE_ID));
-            } else {
-                final String tenantId = getTenantParam(ctx);
-                logger.debug("registering data for device [tenant: {}, device: {}, payload: {}]", tenantId, deviceId, payload);
-                final JsonObject requestMsg = EventBusMessage.forOperation(RegistrationConstants.ACTION_REGISTER)
-                        .setTenant(tenantId)
-                        .setDeviceId((String) deviceId)
-                        .setJsonPayload(payload)
-                        .toJson();
-                sendAction(ctx, requestMsg, getDefaultResponseHandler(ctx,
-                        status -> status == HttpURLConnection.HTTP_CREATED,
-                        response -> response.putHeader(
-                                HttpHeaders.LOCATION,
-                                String.format("/%s/%s/%s", RegistrationConstants.REGISTRATION_ENDPOINT, tenantId, deviceId))));
+            final String deviceId = getDeviceIdParam(ctx);
+            if (deviceId != null) {
+                if (!(deviceId instanceof String)) {
+                    HttpUtils.badRequest(ctx, String.format("'%s' must be a string",
+                            RegistrationConstants.FIELD_PAYLOAD_DEVICE_ID));
+                }
             }
+            final String tenantId = getTenantParam(ctx);
+            logger.debug("registering data for device [tenant: {}, device: {}, payload: {}]", tenantId, deviceId, payload);
+            final JsonObject requestMsg = EventBusMessage.forOperation(RegistrationConstants.ACTION_REGISTER)
+                    .setTenant(tenantId)
+                    .setDeviceId(deviceId)
+                    .setJsonPayload(payload)
+                    .toJson();
+            sendAction(ctx, requestMsg, getDefaultResponseHandler(ctx,
+                    status -> status == HttpURLConnection.HTTP_CREATED,
+                    (response, responsePayload) -> response.putHeader(
+                            HttpHeaders.LOCATION,
+                            String.format("/%s/%s/%s", RegistrationConstants.REGISTRATION_ENDPOINT, tenantId,
```

Comment:
use *getName* instead of `RegistrationConstants.REGISTRATION_ENDPOINT` when creating the *Location* value...